### PR TITLE
Rename `count_min` statistics to `countmin`

### PR DESCRIPTION
**Pull request by JackyWoo**
See https://github.com/ClickHouse/ClickHouse/pull/67013#discussion_r1744346838

### Changelog category (leave one):
- Backward Incompatible Change

### CODE REVIEW

**Comment by JackyWoo** [comment]
@rschu1ze please look at this PR, thanks.

File: docs/en/engines/table-engines/mergetree-family/mergetree.md [hunk 1]
```diff
@@ -1003,16 +1003,16 @@ They can be used for prewhere optimization only if we enable `set allow_statisti
 
     [HyperLogLog](https://en.wikipedia.org/wiki/HyperLogLog) sketches which provide an estimation how many distinct values a column contains.
 
-- `count_min`
+- `CountMin`
```
Comment (rschu1ze) [inline docs/en/engines/table-engines/mergetree-family/mergetree.md:1006]: Names of statistics types are case-insensitive, i.e. both

```sql
CREATE TABLE tab
(
    a LowCardinality(Int64) STATISTICS(minmax)
) [...]
```

and

```sql
CREATE TABLE tab
(
    a LowCardinality(Int64) STATISTICS(MiNmAx)
) [...]
```
 work.

Likewise, names of skipping indexes are case-insensitive. In the [corresponding docs](https://clickhouse.com/docs/en/engines/table-engines/mergetree-family/mergetree#available-types-of-indices), the headings are CamelCase but there is also a "syntax" field where the statistics name is listed lowercase.

Question: Could you please add `Syntax: minmax` etc. to l. 994-1008?

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 077b8239f55163ae25f901924f5a93997d084256 with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/69377/077b8239f55163ae25f901924f5a93997d084256/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/69377/077b8239f55163ae25f901924f5a93997d084256/style_check.html">❌ failure</a></td></tr>
<tbody>
</table>

**Comment by rschu1ze** [comment]
Besides above comment, the PR looks good (thanks).

**Review by rschu1ze — APPROVED** [review]

**Comment by rschu1ze** [comment]
Stateful tests (ubsan, ParallelReplicas): Infra issue?

```
Code: 76. DB::Exception: Cannot lock file /var/run/clickhouse-server/clickhouse-server.pid. Another server instance in same directory is already running. (CANNOT_OPEN_FILE), Stack trace (when copying this message, always include the lines below):

0. std::exception::capture() @ 0x0000000018fa594b
1. ./contrib/llvm-project/libcxx/include/exception:114: Poco::Exception::Exception(String const&, int) @ 0x0000000038102065
2. ./build_docker/./src/Common/Exception.cpp:109: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x00000000259dbd6b
3. DB::Exception::Exception(PreformattedMessage&&, int) @ 0x0000000018f7d136
4. DB::Exception::Exception<String const&>(int, FormatStringHelperImpl<std::type_identity<String const&>::type>, String const&) @ 0x0000000018fa122f
5. ./build_docker/./src/Common/StatusFile.cpp:75: DB::StatusFile::StatusFile(String, std::function<void (DB::WriteBuffer&)>) @ 0x0000000025d16bca
6. ./contrib/llvm-project/libcxx/include/__memory/construct_at.h:35: DB::StatusFile* std::construct_at[abi:v15007]<DB::StatusFile, String, std::function<void (DB::WriteBuffer&)>&, DB::StatusFile*>(DB::StatusFile*, String&&, std::function<void (DB::WriteBuffer&)>&) @ 0x0000000025eab6ab
7. ./contrib/llvm-project/libcxx/include/optional:374: BaseDaemon::initialize(Poco::Util::Application&) @ 0x0000000025ea065d
8. ./build_docker/./programs/server/Server.cpp:402: DB::Server::initialize(Poco::Util::Application&) @ 0x0000000025c7219f
9. ./build_docker/./base/poco/Util/src/Application.cpp:310: Poco::Util::Application::run() @ 0x00000000382a45f3
10. ./build_docker/./programs/server/Server.cpp:396: DB::Server::run() @ 0x0000000025c71f12
11. ./build_docker/./programs/server/Server.cpp:202: mainEntryClickHouseServer(int, char**) @ 0x0000000025c6e285
12. ./build_docker/./programs/main.cpp:248: main @ 0x0000000018f6ebfe
13. ? @ 0x00007ff48724dd90
14. ? @ 0x00007ff48724de40
15. _start @ 0x0000000018f5602e
 (version 24.9.1.2196)
```

**Pull request merged**